### DISCUSSION TITLE
Filtering Data / Do search for

### DISCUSSION THREAD

**Discussion by twilson9034** [Budibase/budibase#3590] [Help]
I have buttons on a Message screen. im trying to figure out how i can hide those button if the user is not part of that team. 

I have the buttons nested in a data provider for the table that contains the teams and team members. All i have is New Data Provider.Schema. Is there a way to drill down further?  I would think we should be able to access that data when trying to complete conditions or filters.

Or someway to do search for and look into that data table. 

I come from bubble and powerapps and Budibase is more "code" then both so if there is a way to do it with javascript or handlebars awesome just be patient with me and ill pick it up.

Overall I would think the condition should go like this> 
![image](https://user-images.githubusercontent.com/94488623/143947956-7729a440-4f50-464c-9309-5a16058990f9.png)

**Comment by twilson9034** [comment]
Ive gotten to be able to filter by putting it in a repeater however its not hiding the component. more work to do.

**Comment by mjashanks** [comment]
Hi @twilson9034 

I have successfully hidden components that do not belong to the `Current User`'s team.

My table structure:
`One Team has many Users (i.e. "members")`
`One Team has many Tasks`

In my design section, I am 
1. displaying all `Tasks`
2. In each task row, I show/hide a Button, depending on whether the `Task` belongs to the same `Team` as the current user

My Structure in `Design` looks like:

![image](https://user-images.githubusercontent.com/3524181/144053854-6e138cb9-71ca-4353-9f42-43c3ded2267a.png)

Notice how the task in the middle has no "Edit" button? That's because my user's team is `Development`, not `Sales`.

My condition on the button looks like:

![image](https://user-images.githubusercontent.com/3524181/144054078-de0594a3-8093-4096-8d19-40f9ce10ee3a.png)

However. I noted that you are also talking about filtering the data based on the current user's team. I think that this has been possible in the past, but may have been broken. The way it should work is defined in this bug report (which I've just created): https://github.com/Budibase/budibase/issues/3596

Basically, you should be using nested `Data Providers`, each using a `Relationship` for its data. We are looking into this.

**Comment by twilson9034** [comment]
@mjashanks I do not have my table structure like yours. Currently my structure is as follows.

Users -> has Name Column ( firstName LastName) The Name column is the table display column.
Team -> has Member Coulmn ( which is the Name ) no relationship in the tables themselves. 

![image](https://user-images.githubusercontent.com/94488623/144111895-6b6abacd-4b83-4a0b-b330-ee0095c8ba25.png)
Team Messages is a data provider with Ministry leadership as the (Team data provider) and then the repeater to show the teams.

This is probably where the bug is like you were saying because i cannot get these buttons to hide when the condition is set
![image](https://user-images.githubusercontent.com/94488623/144112201-c9e082dd-5d8e-4b16-8def-72757c1b1e73.png)
![image](https://user-images.githubusercontent.com/94488623/144112429-47b82a51-7213-4402-901f-42b24a1b4106.png)

**Reply by mjashanks** [reply]
Ah, i see - thanks for the clarification. I just set this up as you described, and it worked for me.

Tables
`User has a Name (Text)`
`Team has a Leader (Text)`

I created 2 `Team` rows, only one of them has its `Leader` that matches the value in my user's `Name` column.

I then created the condition below. Notice how the condition doe not seem to be applied. This is because we **always** show a component when its selected in the designer preview.

![image](https://user-images.githubusercontent.com/3524181/144218561-dac30ac1-7611-4133-bd36-842217af4c5b.png)

When I load my app, it looks like:

![image](https://user-images.githubusercontent.com/3524181/144218780-1b74a71d-95f6-42bf-a30d-310862c860e6.png)

**Reply by twilson9034** [reply]
@mjashanks would you mine taking a look at this. I have it working on Screen Admin with a card block but only if i use the Name as a value. 

On screen Team Messages they wont work properly. 

May have been an import issue, I looked at my user table and the data was not there.

**Reply by twilson9034** [reply]
Works using the new card block.
